### DISCUSSION TITLE
Can't use API key with LocationService

### DISCUSSION THREAD

**Discussion by jschaedl** [aws/aws-sdk-php#2749] [Q&A]
I'm trying to make a `SearchPlaceIndexForTextRequest` to the `LocationService` by using an API Key. Here is my (simplified) code:

```php
$sdk = new Sdk();

$locationServiceClient = $sdk->createLocationService([
    'region' => 'eu-central-1',
    'version' => 'latest',
    'credentials' => false
]);

try {
    $result = $locationServiceClient->searchPlaceIndexForText([
        'FilterCountries' => ['DEU'],
        'IndexName' => 'api-evaluation-only-here',
        'Key' => 'my key',
        'Text' => 'brandenburger tor'
    ]);
} catch (\Exception $exception) {
    dump($exception->getMessage());
}
```

This results in the following exception:

```
Error executing "SearchPlaceIndexForText" on "https://places.geo.eu-central-1.amazonaws.com/places/v0/indexes/api-evaluation-only-here/search/text"; AWS HTTP error: Client error: `POST https://places.geo.eu-central-1.amazonaws.com/places/v0/indexes/api-evaluation-only-here/search/text` resulted in a `403 Forbidden` response:
{"message":"Missing Authentication Token"}
 MissingAuthenticationTokenException (client): Missing Authentication Token - {"message":"Missing Authentication Token"}
```

The documentation says I can pass a `Key` along with the method arguments: https://docs.aws.amazon.com/aws-sdk-php/v3/api/api-location-2020-11-19.html#searchplaceindexfortext but it is not added as a query parameter.

Any idea what I am doing wrong or is this a bug?

**Comment by janklan** [comment]
I have the same question. The Location service no longer requires custom Place Indexes or Maps to be provisioned - it's supposed to be enough to use an API key, but the SDK didn't seem to have caught up yet - the [new](https://docs.aws.amazon.com/location/latest/developerguide/how-to-search-address.html) (`v2/search-text` etc) API endpoints are nowhere to be found in https://github.com/aws/aws-sdk-php/blob/master/src/data/location/2020-11-19/api-2.json
